Automattic/vip-go-mu-plugins
# Add mirrored search requests ability for evaluation

## Description

For running tests on VIP Search, add code to run VIP Search-powered queries in parallel with the existing implementation.

## Checklist

Please make sure the items below have been covered before requesting a review:

- n/a This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Example:

1. Check out PR.
1. Make sure VIP Search is enabled and query integration is _off_ (`VIP_ENABLE_ELASTICSEARCH_QUERY_INTEGRATION` is not defined)
1. Enable request duplication: `define( 'VIP_ENABLE_SEARCH_QUERY_MIRRORING', true );`
1. Do a search - `?s=test`.
1. Check origin request logs to verify we log a request for `?s=test&es=true` (should have a UA of `VIP Search Query Mirror; <siteurl>`)

Then, remove constant and enable the mirroring option: `wp option update vip_enable_search_query_mirroring 1` and repeat.

To clean up, remove the option `wp option delete vip_enable_search_query_mirroring`
